### ISSUE TITLE
`amplify:<function-name>` scripts that install devdependencies for build are installing production dependencies due to `NODE_ENV=production`

### ISSUE THREAD

**Issue by chrisl777** [aws-amplify/amplify-cli#11785]
### How did you install the Amplify CLI?

yarn

### If applicable, what version of Node.js are you using?

14.19.1

### Amplify CLI Version

10.6.1

### What operating system are you using?

Mac

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No manual changes made

### Describe the bug

Without any changes to our configuration, within the last week, the Amplify console build has recently stopped including the executable code for TypeScript Lambdas into the deployed Lambdas. We have been running under the same Lambda configuration for over a year. 

I noticed b/c our CloudWatch logs for the Lambdas were showing an error: 

```
"errorType": "Runtime.HandlerNotFound",
"errorMessage": "index.handler is undefined or not exported"
```

The result of the Amplify console build that is stored in the deployment bucket in S3 is just the package.json, the compiled index.js is not included in the S3 deployment. 

Running the build script locally with `yarn amplify:{LambdaName}` does output an index.js to the src/ folder within the Lambda. 

Note that if we run `amplify push` manually from the command line, then the Lambdas do get created properly and the resulting executable index.js files are included into the deployment. 

Here's the folder structure for each Lambda (within /amplify/backend/function/...): 
```
-dist/ (not committed, resulting build folder)
-lib/
--source files (.ts) 
-src/
--output index.js 
-{LambdaName}-cloudformation-template.json 
-package.json 
-tsconfig.json 
-webpack.config.js
-yarn.lock 
```

The tsconfig.json files: 
```
{
  "compilerOptions": {
    "lib": [
      "es2017",
      "DOM"
    ], 
    "target": "es2017", 
    "module": "commonjs", 
    "moduleResolution": "node",     
    "baseUrl": "./",
    "sourceMap": true,
    "esModuleInterop": true,
    "types": [
      "node",
      "jest"
    ],
  },
  "exclude": [
    "node_modules",
    "**/*.test.ts"
  ]
} 
```

The build script in our root package.json: 
```
"amplify:{LambdaName}": "cd amplify/backend/function/{LambdaName} && yarn install && webpack-cli && cd -",
```

I suspect that we may need to add `"outDir": "./src"` to our tsconfig.json, since that is where the index.js files end up, but I'm not sure, since it is working with a local `amplify push`. 

### Expected behavior

We expect that the compiled index.js files will be included in the deployment buckets (and thus deployed as part of the deployed Lambda files). 

### Reproduction steps

1. Create TypeScript Lambdas per the docs: https://docs.amplify.aws/cli/function/build-options/
2. Set up Amplify console CI/CD process. 
3. Push code to repo 
4. Observe the resulting built Lambdas, the folders do not contain index.js files. 

### Project Identifier

ac2fa547ce795a99c33b7773e2d59f15

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by chrisl777** [comment]
@josefaidt Any update on this bug?

**Comment by josefaidt** [comment]
Hey @chrisl777 this fix is pending release and should be available in the next version of the CLI

**Comment by josefaidt** [comment]
This fix was released with Amplify CLI 10.7.0 🎉 

closing issue
